app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt

```diff
@@ -232,10 +244,14 @@ class HomePresenter(
 
     private fun handleUserTokensLoaded(userTokens: List<Token.Active>) {
         launch {
-            val ethereumTokens = try {
-                ethereumRepository.loadWalletTokens()
-            } catch (throwable: Throwable) {
-                Timber.e(throwable, "Error on loading ethereumTokens")
+            val ethereumTokens = if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+                try {
+                    ethereumRepository.loadWalletTokens()
+                } catch (throwable: Throwable) {
+                    Timber.e(throwable, "Error on loading ethereumTokens")
+                    emptyList()
+                }
```

Comment:
```suggestion
                runCatching { ethereumRepository.loadWalletTokens() } 
                    .onFailure { Timber.e(it, "Error on loading ethereumTokens") }
                    .getOrDefault(emptyList())
```
how's that?

Reply:
I like the first variant the most. 
Don't really like chains